service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -436,15 +441,36 @@ public void editSite(JWTData userData, int siteId, EditSiteRequest editSiteReque
   }
 
   @Override
-  public void addSites(JWTData userData, AddSitesRequest addSitesRequest) {
+  public void addSites(JWTData userData, String addSitesRequest) {
     assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
 
-    addSitesRequest
-        .getSites()
-        .forEach(
-            newSite -> {
-              addSite(userData, newSite);
-            });
+    List<AddSiteRequest> addSiteRequests = this.parseCSVString(addSitesRequest);
+
+    addSiteRequests.forEach(siteRequest -> addSite(userData, siteRequest));
+  }
+
+  /**
+   * Parses the given CSV string containing data on a site and site entry to a list of
+   * AddSiteRequests.
+   *
+   * @param sitesCSV CSV string to parse
+   * @throws HandledException if the given CSV string cannot be parsed properly
+   * @return the parsed list of AddSiteRequests
+   */
+  private List<AddSiteRequest> parseCSVString(String sitesCSV) throws HandledException {
+    try {
+      CsvMapper mapper = new CsvMapper();
+      CsvSchema schema = CsvSchema.emptySchema().withHeader();
+      MappingIterator<AddSiteRequest> sitesIterator =
+          mapper.readerFor(AddSiteRequest.class).with(schema).readValues(sitesCSV);
+      List<AddSiteRequest> addSiteRequests = sitesIterator.readAll();
+      for (AddSiteRequest addSiteRequest : addSiteRequests) {
+        addSiteRequest.validate();
+      }
```

Comment:
⛏️ Could we use the same `List.forEach` method from line 449 to make this block a little more readable? Maybe something like:
```suggestion
      addSiteRequests.forEach(siteRequest -> siteRequest.validate());
```